msg/versioned/ArmingCheckReply.msg

```diff
@@ -1,34 +1,42 @@
+# The message is not used by internal/FMU components, as their mode requirements are known at compile time.
+
 uint32 MESSAGE_VERSION = 0
 
-uint64 timestamp # time since system start (microseconds)
+uint64 timestamp # [us] Time since system start.
 
-uint8 request_id
-uint8 registration_id
+uint8 request_id # Id of ArmingCheckRequest for which this is a response.
+uint8 registration_id # Id of external component emitting this response.
 
-uint8 HEALTH_COMPONENT_INDEX_NONE = 0
+uint8 HEALTH_COMPONENT_INDEX_NONE = 0 # Index of health component for which this response applies.
 
-uint8 health_component_index      # HEALTH_COMPONENT_INDEX_*
-bool health_component_is_present
-bool health_component_warning
-bool health_component_error
+uint8 health_component_index # [@enum HEALTH_COMPONENT_INDEX]
+bool health_component_is_present # Unused. Intended for use with health events interface (health_component_t in events.json).
+bool health_component_warning # Unused. Intended for use with health events interface (health_component_t in events.json).
+bool health_component_error # Unused. Intended for use with health events interface (health_component_t in events.json).
 
-bool can_arm_and_run              # whether arming is possible, and if it's a navigation mode, if it can run
+bool can_arm_and_run # True for navigation mode components if the vehicle meets all the requirements to arm in the mode (or switch to the mode if it is already active). For non-mode components, such as remote IDs, whether the component can be armed.
 
-uint8 num_events
+uint8 num_events # Number of queued failure messages (Event) in the events field.
 
-Event[5] events
+Event[5] events # Arming failure reasons (Queue of events to report to GCS).
 
 # Mode requirements
-bool mode_req_angular_velocity
-bool mode_req_attitude
-bool mode_req_local_alt
-bool mode_req_local_position
-bool mode_req_local_position_relaxed
-bool mode_req_global_position
-bool mode_req_mission
-bool mode_req_home_position
-bool mode_req_prevent_arming
-bool mode_req_manual_control
-
-
-uint8 ORB_QUEUE_LENGTH = 4
+bool mode_req_angular_velocity # Requires angular velocity estimate (e.g. from gyroscope).
+bool mode_req_attitude # Requires an attitude estimate.
+bool mode_req_local_alt # Requires a local altitude estimate.
+bool mode_req_local_position # Requires a local position estimate.
+bool mode_req_local_position_relaxed # Requires a more relaxed global position estimate.
+bool mode_req_global_position # Requires a global position estimate.
+bool mode_req_mission # Requires an uploaded mission.
+bool mode_req_home_position # Requires a home position (such as RTL/Return mode).
+bool mode_req_prevent_arming # Prevent arming (such as in Land mode).
+bool mode_req_manual_control # Requires a manual controller
+
+uint8 ORB_QUEUE_LENGTH = 4 # 
```

Comment:
You said:

> In uorb, the default is to have a single buffer which is overwritten when multiple samples get published too quickly. This allows access to multiple samples by the subscriber.

So is this uorb magic like the version number? - i.e. a uint8 with the name `ORB_QUEUE_LENGTH` always does some magic under the hood to create a queue, such that if you subscribe to a topic you'll be able to get up to four instances of the message if you're slow at reading them, rather than the last to arrive?

If so, then this isn't documented in https://docs.px4.io/main/en/middleware/uorb.html#message-definitions - and we should cover this syntax. Thoughts?

Reply:
Yes you are correct, it's generic, and should be documented there.
If set, it has to be a power of 2 (so 2, 4, ...)